File: frontend/types/loan.ts
export type LoanStatus = 'Pending' | 'Approved' | 'Rejected' | 'Active';

export type LoanType = 'Fixed' | 'Decreasing';

export type PaymentScheduleStatus = 'Pending' | 'Paid';

export interface LoanResponse {
  id: string;
  userId: string;
  amount: number;
  termMonths: number;
  annualInterestRate: number;
  loanType: LoanType;
  status: LoanStatus;
  monthlyPayment: number;
  createdAt: string;
  updatedAt: string;
}

export interface PaymentInstallment {
  paymentNumber: number;
  dueDate: string;
  totalPayment: number;
  principal: number;
  interest: number;
  remainingBalance: number;
}

export interface SimulateLoanResponse {
  amount: number;
  termMonths: number;
  annualInterestRate: number;
  monthlyPayment: number;
  loanType: LoanType;
  schedule: PaymentInstallment[];
}

export interface PaymentScheduleResponse {
  paymentNumber: number;
  dueDate: string;
  totalPayment: number;
  principal: number;
  interest: number;
  remainingBalance: number;
  status: PaymentScheduleStatus;
}

export interface SimulateLoanRequest {
  amount: number;
  termMonths: number;
  loanType?: LoanType;
}

export interface CreateLoanRequest {
  amount: number;
  userId?: string;
  termMonths: number;
  loanType?: LoanType;
  monthlyIncome?: number;
}

File: src/Core/FinTech.Domain/Entities/Loan.cs
namespace FinTech.Domain.Entities;

public class Loan : BaseEntity<Guid>
{
    public string UserId { get; set; } = null!;
    public decimal Amount { get; set; }
    public int Term { get; set; } // Months
    public decimal InterestRate { get; set; } // TEA (Annual Effective Rate)
    public LoanType LoanType { get; set; }
    public LoanStatus Status { get; set; }
    public decimal MonthlyPayment { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public virtual ICollection<PaymentSchedule> PaymentSchedules { get; set; } = [];
    public virtual ICollection<Transaction> Transactions { get; set; } = [];
}

public enum LoanStatus
{
    Pending,
    Approved,
    Rejected,
    Active
}

public enum LoanType
{
    Fixed,
    Decreasing
}
